app/Providers/AppServiceProvider.php

```diff
@@ -35,5 +37,25 @@ function ($app) use ($socialite) {
                 return $socialite->buildProvider(VATSIMUKProvider::class, $config);
             }
         );
+
+        Http::macro('ukcp', function ($endpoint, $method = 'get', $body = []) {
+            $apiToken = config('services.vatsim_uk_controller_api.token');
+            $bodyRequiredMethods = ['post', 'patch'];
+            $requestedMethodRequiresBody = in_array(strtolower($method), $bodyRequiredMethods);
+
+            if ($requestedMethodRequiresBody && $body === []) {
+                throw new InvalidArgumentException(sprintf('Body required when making request using %s', implode(separator: ', ', array: $bodyRequiredMethods)));
+            }
+
+            $baseHttpObject = Http::withHeaders([
+                'Authorization' => "Bearer {$apiToken}",
+            ]);
```

Comment:
Can simplify if desired:
```suggestion
            $baseHttpObject = Http::withToken($apiToken);
```

Reply:
Done!